tests/test_Register.py

```diff
@@ -0,0 +1,231 @@
+                                      inData.templateExp.getBBox(afwImage.LOCAL))
+        warpedSources = task.warpSources(inData.inputSources, results.wcs, inData.templateExp.getWcs(),
+                                         inData.templateExp.getBBox(afwImage.LOCAL))
+
+        self.showImage(warpedExp, warpedSources, "Aligned", 3)
+        return Struct(warpedExp=warpedExp, warpedSources=warpedSources, matches=results.matches,
+                      wcs=results.wcs, task=task)
+
+    def assertRegistered(self, inData, outData, bad=set()):
+        """Assert that the registration task is registering images"""
+        xTemplate = np.array([x for i, x in enumerate(inData.xTemplate) if i not in bad])
+        yTemplate = np.array([y for i, y in enumerate(inData.yTemplate) if i not in bad])
+        alignedArray = outData.warpedExp.getMaskedImage().getImage().getArray()
+        self.assertTrue((alignedArray[yTemplate, xTemplate] == 1.0).all())
+        for dx in (-1, 0, +1):
+            for dy in range(-1, 0, +1):
+                # The density of points is such that I can assume that no point is next to another.
+                # The values are not quite zero because the "image" is undersampled, so we get ringing.
+                self.assertTrue((alignedArray[yTemplate+dy, xTemplate+dx] < 0.1).all())
+
+        xAligned = np.array([x for i, x in enumerate(outData.warpedSources["center_x"]) if i not in bad])
+        yAligned = np.array([y for i, y in enumerate(outData.warpedSources["center_y"]) if i not in bad])
+        self.assertAlmostEqual((xAligned - xTemplate).mean(), 0, 8)
+        self.assertAlmostEqual((xAligned - xTemplate).std(), 0, 8)
+        self.assertAlmostEqual((yAligned - yTemplate).mean(), 0, 8)
+        self.assertAlmostEqual((yAligned - yTemplate).std(), 0, 8)
+
+    def assertMetadata(self, outData, numRejected=0):
+        """Assert that the registration task is populating the metadata"""
+        metadata = outData.task.metadata
+        self.assertEqual(metadata.get("MATCH_NUM"), self.numSources)
+        self.assertAlmostEqual(metadata.get("SIP_RMS"), 0.0)
+        self.assertEqual(metadata.get("SIP_GOOD"), self.numSources-numRejected)
+        self.assertEqual(metadata.get("SIP_REJECTED"), numRejected)
+
+    def testRegister(self):
+        """Test image registration"""
+        inData = self.create()
+        outData = self.runTask(inData)
+        self.assertRegistered(inData, outData)
+        self.assertMetadata(outData)
+
+    def testRejection(self):
+        """Test image registration with rejection"""
+        inData = self.create()
+
+        # Tweak a source to have a bad offset
+        badIndex = 111
+
+        coordKey = inData.inputSources[badIndex].getTable().getCoordKey()
+        centroidKey = inData.inputSources[badIndex].getTable().getCentroidKey()
+        x, y = float(inData.xInput[badIndex] + 0.01), float(inData.yInput[badIndex] - 0.01)
```

Comment:
Should the `0.01` here be some multiple of `self.pixelscale`? Otherwise, it's not clear why `0.01` is the right value.